File: other/ABC184D.py
A, B, C = map(int, input().split())

dp = [[[0] * 101 for _ in range(101)] for _ in range(101)]


def rec(a, b, c):
    if dp[a][b][c]:
        return dp[a][b][c]
    if a == 100 or b == 100 or c == 100:
        return 0
    ret = 0
    ret += (rec(a + 1, b, c) + 1) * a / (a + b + c)
    ret += (rec(a, b + 1, c) + 1) * b / (a + b + c)
    ret += (rec(a, b, c + 1) + 1) * c / (a + b + c)
    dp[a][b][c] = ret
    return ret


print("{:.9f}".format(rec(A, B, C)))

"""
愚直なら3^(3N-A-B-C)　んなこたあない

dpっぽい

解説後
漸化式を書くべき
遷移前・遷移後で期待値の関係性が存在するはず
"""

File: rust/src/contest/abc184/abc184_d.rs
#![allow(clippy::let_unit_value)]
#![allow(non_snake_case)]
use std::collections::HashMap;

use proconio::input;

struct ExpDP {
    m: HashMap<(usize, usize, usize), f64>,
}
impl ExpDP {
    fn solve(a: usize, b: usize, c: usize) -> f64 {
        ExpDP { m: HashMap::new() }.rec(a, b, c)
    }

    fn rec(&mut self, a: usize, b: usize, c: usize) -> f64 {
        if self.m.contains_key(&(a, b, c)) {
            return self.m[&(a, b, c)];
        }
        if a == 100 || b == 100 || c == 100 {
            return 0_f64;
        }
        let mut ret = 0_f64;
        let z = a as f64 + b as f64 + c as f64;
        if a != 0 {
            ret += a as f64 / z * self.rec(a + 1, b, c);
        }
        if b != 0 {
            ret += b as f64 / z * self.rec(a, b + 1, c);
        }
        if c != 0 {
            ret += c as f64 / z * self.rec(a, b, c + 1);
        }

        self.m.insert((a, b, c), ret + 1.0);

        ret + 1.0
    }
}

fn main() {
    input! {
        A:usize,
        B:usize,
        C:usize
    }
    let ret = ExpDP::solve(A, B, C);
    println!("{:.10}", ret)
}
